src/wp-includes/class-walker-nav-menu.php

```diff
@@ -276,4 +307,24 @@ public function end_el( &$output, $data_object, $depth = 0, $args = null ) {
 		$output .= "</li>{$n}";
 	}
 
+	/**
+	 * Builds an HTML attributes string based on an array of key value pairs
+	 * Empty values are ignored
+	 *
+	 * @since 6.1.2
+	 *
+	 * @param      array   $atts   The atts
+	 * @return     string  HTML attributes string
+	 */
+	protected function build_atts( $atts = array() ) {
+		$attributes = '';
+		foreach ( $atts as $attr => $value ) {
+			if ( is_scalar( $value ) && '' !== $value && false !== $value ) {
+				$value       = ( 'href' === $attr ) ? esc_url( $value ) : esc_attr( $value );
+				$attributes .= ' ' . $attr . '="' . $value . '"';
```

Comment:
Curious about `(bool) true` values here.

`(bool) false` are skipped, but `(bool) true` will be accepted. That is, `$attr = true` will become `attr="1"`.

Is this the intended/desired behaviour, or should `(bool) true` be treated as a boolean attribute and only `esc_attr( $attr )` be appended to `$attributes`?

Reply:
I mainly replicated what was already in use in this file to avoid creating breaking changes as much as possible. Any scalar value can be easily transformed into a string by PHP, some people might already be relying in their code on ```(bool) true``` becoming (string) "1" so I wouldn't want to step into that for now.